### PR TITLE
Refactoring template preprocessor

### PR DESCRIPTION
**Pull request by warrenbailey**
### What is the context of this PR?

Refactoring template preprocessor, attempting to simplify and removing unnecessary code. The idea being that now we know what's being used and hopefully this means we can rewrite the rendering to be a lot simpler.
### How to review
1. Check out this branch
2. Confirm all the tests still pass
3. Complete the star wars and MCI surveys check nothing is broken or missing. Pay particular attention to the meta data fields
- [ ] Do all commits have sensible commit messages?
- [ ] Is all new code unit tested?
- [ ] Are there integration tests if needed?
- [ ] Is there sufficient logging?
- [ ] Is there documentation or is the code self-describing?
### Who worked on the PR

@warrenbailey

### CODE REVIEW

**Comment by weapdiv-david** [comment]
LGTM :+1:!  Lewis?

File: app/templating/template_pre_processor.py [hunk 1]
```diff
@@ -1,47 +1,29 @@
...
-        # get the group
-        self._current_group = self._current_block.container
-
-        piping_context = self._build_piping_context()
-
-        self._plumber = Plumber(piping_context)
-
-    def get_template_name(self):
-        known_templates = {
+KNOW_TEMPLATES = {
```
Comment (LJBabbage) [inline app/templating/template_pre_processor.py]: Known?

**Comment by LJBabbage** [comment]
Looks good and works as expected

**Comment by warrenbailey** [comment]
Rebased and ready for merging

**Comment by LJBabbage** [comment]
AttributeError: 'TemplatePreProcessor' object has no attribute 'questionnaire_manager'

**Pull request merged**